src/snowflake/snowpark/dataframe.py

```diff
@@ -1133,8 +1139,11 @@ def select(
         for e in exprs:
             if isinstance(e, Column):
                 names.append(e._named())
+                ast.sp_dataframe_select__columns.cols.append(e.ast)
             elif isinstance(e, str):
-                names.append(Column(e)._named())
+                col = Column(e)
+                names.append(col._named())
+                ast.sp_dataframe_select__columns.cols.append(col.ast)
```

Comment:
This is inefficient. Since the user passes in a string in this case, you're free to construct a new column expr from scratch without the inefficient copying.

This will naturally get refactored due to the previous comment though.

If this were to result in cleaner code, feel free to reimplement the logic, since in Phase 1 we will remove a lot of the existing complexity (we're preserving it for now so that this codebase can be used for Phase 0).

Reply:
More info at https://protobuf.dev/reference/python/python-generated/#repeated-message-fields